src/App.jsx

```diff
@@ -14,33 +17,104 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const SORT_FIELD_ALPHABET = 'alphabet';
+const SORT_FIELD_LENGTH = 'length';
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+function getSortedGoods(goods, { sortField, isReverse }) {
+  const sortedGoods = [...goods];
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  if (sortField) {
+    sortedGoods.sort((a, b) => {
+      switch (sortField) {
+        case SORT_FIELD_ALPHABET:
+          return a.localeCompare(b);
+        case SORT_FIELD_LENGTH:
+          return a.length - b.length;
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+        default:
+          return 0;
+      }
+    });
+  }
+
+  if (isReverse) {
+    sortedGoods.reverse();
+  }
+
+  return sortedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState('');
+  const [isReverse, setIsReverse] = useState(false);
+
+  const sortedGoods = getSortedGoods(goodsFromServer, { sortField, isReverse });
 
-    <ul>
-      <li data-cy="Good">Dumplings</li>
-      <li data-cy="Good">Carrot</li>
-      <li data-cy="Good">Eggs</li>
-      <li data-cy="Good">Ice cream</li>
-      <li data-cy="Good">Apple</li>
-      <li data-cy="Good">...</li>
-    </ul>
-  </div>
-);
+  return (
+    <div className="section content">
+      <div className="buttons">
+        <button
+          type="button"
+          className={cn({
+            'button': true,
+            'is-info': true,
+            'is-light': sortField !== SORT_FIELD_ALPHABET,
+          })}
```

Comment:
Default classes do not need to be checked (Do the same in other buttons)

```suggestion
          className={cn('button', 'is-info', {
            'is-light': sortBy !== SORT_FIELD_ALPHABET,
          })}
```